### PR TITLE
Dupdup

### PR DESCRIPTION
**Pull request by hardening**
Contains the handling of strdup by @bmiklautz. And my own fixes.
The NLA part should conflict with https://github.com/FreeRDP/FreeRDP/pull/2702.

### CODE REVIEW

**Comment by bmiklautz** [comment]
My strdup changes/commits are still work in progress and not ready for a PR yet.

File: channels/audin/client/opensles/audin_opensl_es.c [hunk 1]
```diff
@@ -161,7 +161,9 @@ static void audin_opensles_free(IAudinDevice* device)
 
 static BOOL audin_opensles_format_supported(IAudinDevice* device, audinFormat* format)
 {
+#ifdef WITH_DEBUG_DVC
 	AudinOpenSLESDevice* opensles = (AudinOpenSLESDevice*) device;
+#endif
 	
 	DEBUG_DVC("device=%p, format=%p", opensles, format);
```
Comment (realjiangms) [inline channels/audin/client/opensles/audin_opensl_es.c]: Will we have compile error if WITH_DEBUG_DVC is not set?

Comment (bmiklautz) [inline channels/audin/client/opensles/audin_opensl_es.c]: @realjiangms nope there shouldn't be an error. Have a look to
channels/audin/client/audin_main.h around line 35.

**Comment by bmiklautz** [comment]
Superseded by #2708

**Pull request closed without merging**